src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -3128,4 +3145,123 @@ public Member getMember()
             return member;
         }
     }
+
+    /**
+     * Metadata about the interaction, including the source of the interaction and relevant server and user IDs.
+     *
+     * @see Message#getInteractionMetadata()
+     */
+    class InteractionMetadata implements ISnowflake
+    {
+        private final long id;
+        private final int type;
+        private final User user;
+        private final IntegrationOwners integrationOwners;
+        private final Long originalResponseMessageId;
+        private final Long interactedMessageId;
+        private final InteractionMetadata triggeringInteraction;
+
+        public InteractionMetadata(long id, int type, User user, IntegrationOwners integrationOwners, Long originalResponseMessageId, Long interactedMessageId, InteractionMetadata triggeringInteraction)
+        {
+            this.id = id;
+            this.type = type;
+            this.user = user;
+            this.integrationOwners = integrationOwners;
+            this.originalResponseMessageId = originalResponseMessageId;
+            this.interactedMessageId = interactedMessageId;
+            this.triggeringInteraction = triggeringInteraction;
+        }
+
+        @Override
+        public long getIdLong()
+        {
+            return id;
+        }
+
+        /**
+         * The raw interaction type.
+         * <br>It is recommended to use {@link #getType()} instead.
+         *
+         * @return The raw interaction type
+         */
+        public int getTypeRaw()
+        {
+            return type;
+        }
+
+        /**
+         * The {@link net.dv8tion.jda.api.interactions.InteractionType} for this interaction.
+         *
+         * @return The {@link net.dv8tion.jda.api.interactions.InteractionType} or {@link net.dv8tion.jda.api.interactions.InteractionType#UNKNOWN}
+         */
+        @Nonnull
+        public InteractionType getType()
+        {
+            return InteractionType.fromKey(type);
+        }
+
+        /**
+         * The {@link User} who caused this interaction.
+         *
+         * @return The {@link User}
+         */
+        @Nonnull
+        public User getUser()
+        {
+            return user;
+        }
+
+        /**
+         * Returns the integration owners of this interaction, which depends on how the app was installed.
+         *
+         * <p>This will be {@code null} if the command is installed <b>only</b> {@link IntegrationType#GUILD_INSTALL on guilds},
+         * regardless of if the command is on the global scope, or on a guild:
+         * <ul>
+         *     <li>❌ : Global command with {@link IntegrationType#GUILD_INSTALL} (and nothing else)</li>
+         *     <li>❌ : Guild command with {@link IntegrationType#GUILD_INSTALL} (and nothing else)</li>
+         *     <li>✔️ : Any other case</li>
+         * </ul>
+         *
+         * @return The integration owners of this interaction, or {@code null}
+         */
+        @Nonnull
```

Comment:
If its nullable? Why is it marked as Nonnull?

Reply:
Thanks for the reminder, fixed